### ISSUE TITLE
OnPopupClose handler

### ISSUE THREAD

**Issue by dopt** [tcampb/react-calendly#82]
Hi, just wondering do we have any handling for **onClose** action if create a custom button that triggers a pop-up scheduler, e.g: after event schedule, and user click the "X" button to close the modal, I need a handler on that

**Comment by tcampb** [comment]
Hi @dopt - 

Unfortunately, there is not currently an easy way to add a custom **onClose** method to the popup modal that is created when you use `openPopupWidget`. 

To customize the `onClose` method, you would need to replace the event listener that is attached to the "X" button:

```typescript
import { openPopupWidget, closePopupWidget } from "react-calendly";

const onClick = () => {
  openPopupWidget({ url: 'https://calendly.com/acmesales' })
  
  const oldCloseButton = document.querySelector('.calendly-popup-close')
  const newCloseButton = oldCloseButton.cloneNode(true);
  
  oldCloseButton.parentNode.replaceChild(oldCloseButton, newCloseButton);
  
  newCloseButton.addEventListener('click', () => {
    console.log('custom close button')
    closePopupWidget()
  });
}
```

**Comment by tcampb** [comment]
@dopt  - this feature is now available for testing in version [`3.0.0-beta.1`](https://github.com/tcampb/react-calendly/tree/v3.0.0-beta.1), which can be installed using the following command:

```bash
npm i react-calendly@3.0.0-beta.1
```

This new version exports the `PopupModal` component which includes an `onModalClose` property. The `onModalClose` function is run when the user clicks the **X** icon.

```tsx
import { PopupModal } from "react-calendly";

class CustomButtonExample extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      isOpen: false,
    };
  }

  render() {
    return (
      <div>
        <button
          style={{ display: "block", margin: "0 auto" }}
          onClick={() => this.setState({ isOpen: true })}
        >
          Custom Button
        </button>
        <PopupModal
          url="https://calendly.com/acmesales"
          pageSettings={this.props.pageSettings}
          utm={this.props.utm}
          prefill={this.props.prefill}
          onModalClose={() => this.setState({ isOpen: false })}
          open={this.state.isOpen}
          /*
           * react-calendly uses React's Portal feature (https://reactjs.org/docs/portals.html) to render the popup modal. As a result, you'll need to
           * specify the rootElement property to ensure that the modal is inserted into the correct domNode.
           */
          rootElement={document.getElementById("root")}
        />
      </div>
    );
  }
}
```
